src/resolver.rs

```diff
@@ -1279,17 +1364,68 @@ fn resolve_third_party<'a>(
     }
 
     // Reject forbidden packages (violations)
-    //
-    // FIXME: should the "local" audit have a mechanism to override foreign forbids?
     for (violation_source, violation_entry) in &violation_nodes {
+        // Ok this is kind of weird. We want to reject any audits which contain any of these criteria.
+        // Normally we would slap all the criteria in this entry into a set and do some kind of set
+        // comparison, but that's not quite right. Here are the cases we want to work:
+        //
+        // * violation: safe-to-deploy, audit: safe-to-deploy -- ERROR!
+        // * violation: safe-to-deploy, audit: safe-to-run    -- OK!
+        // * violation: safe-to-run,    audit: safe-to-deploy -- ERROR!
+        // * violation: [a, b],         audit: [a, c]         -- ERROR!
+        //
+        // The first 3 cases are correctly handled by audit.contains(violation)
+        // but the last one isn't. I think the correct solution to this is to
+        // *for each individual entry in the violation* do audit.contains(violation).
+        // If any of those queries trips, then it's an ERROR.
+        //
+        // Note that this would also more correctly handle [safe-to-deploy, safe-to-run]
+        // as a violation entry because it would effectively become safe-to-run instead
+        // of safe-to-deploy, which is the correct and desirable behaviour!
+        //
+        // So here we make a criteria set for each entry in the violation.
```

Comment:
Had to read this 3 times, but I'm pretty sure you're right and this should work.